src/org/kie/jenkins/MavenCommand.groovy

```diff
@@ -23,9 +27,19 @@ def class MavenCommand {
 
     def run(String goals) {
         def cmdBuilder = new StringBuilder("mvn -B")
-        if(this.settingsXmlPath) {
-            cmdBuilder.append(" -s ${this.settingsXmlPath}")
+
+        // Retrieve settings file from id if given
+        String settingsFile = this.settingsXmlPath
```

Comment:
I changed to dynamic generation of the settings file because this would create conflict if we don't